### ISSUE TITLE
Globnode under quoted arithmetic links produces error

### ISSUE THREAD

**Issue by Eman22S** [opencog/atomspace#2583]
Glob nodes found under Quote and Unquote link for arithmetic links produced an error which said cannot create an arithmetic link which has a list link inside with the matched glob items. For instance
if we have this Link in our atomspace 
```
(PlusLink
   (Number 1)
   (Number 2)
   (Number 10))

```

and we have the following BindLink

```
(define test
 (BindLink
      (PlusLink
         (Glob "$op")
         (Number 10))
 
      (Quote
      (PlusLink
        (Unquote (Glob "$op"))
         (Number 10)))))
```
This error message is thrown
```
Throw to key `C++-EXCEPTION' with args `("cog-execute!" "Invalid Atom syntax: (PlusLink\n  (ListLink\n    (NumberNode \"1\") ; [2cf08d6d543d00ff][1]\n    (NumberNode \"2\") ; [246c4657667bd0d2][1]\n  ) ; [b1e76594cfa78787][-1]\n  (NumberNode \"10\") ; [2403db657e23aa8b][1]\n) ; [876f461a096d9fad][-1] (/home/user/atomspace/opencog/atoms/base/ClassServer.cc:138)\nFunction args:\n((BindLink\n   (PlusLink\n      (GlobNode \"$op\")\n      (NumberNode \"10\")\n   )\n   (QuoteLink\n      (PlusLink\n         (UnquoteLink\n            (GlobNode \"$op\")\n         )\n         (NumberNode \"10\")\n      )\n   )\n)\n)")'.
```
I have already looked into the problem and believe that  adding these lines of codes shall fix it. https://github.com/Eman22S/atomspace/commit/6eca0b3e95597c08861bd68dafe3a7e4b588ee11#diff-58088ec4fa8343177d9b767a17a60d58R512. I have also written testcases  for it and they all pass.

**Comment by linas** [comment]
> Quote

I guess you are trying to prevent the Plus from being executed?  You might wish to use the `DontExecLink` instead.  Almost no one ever uses the `DontExecLink` so it might be buggy. I dunno.

>  these lines of code

I'm fairly certain there is a simpler fix, I'll try to find that.

> I have also written testcases for it

It would be great if you made a pull req for the test-cases.  Make sure they are indented correctly! (the indentation is broken on the diff you attached)

**Comment by linas** [comment]
Closing; fixed in  #2586

**Comment by linas** [comment]
p.s. thank you for exploring this code. Yes, the patch you provided might fix the problem, but I created a different patch, mostly because I'm more familiar with the code, and know what it should have done.  There's a lot of existing code to handle quoting and unquoting; it should have already done the right thing. ... and it did, for variables.  However, globs are handled in a fundamentally different way, and so unquoting of globs was never tested before. I picked a different fix, because the unquoting-of-globs was a generic problem, and not specific to the arithmetic link.  (I didn't mean to stomp all over your work, or reject it, even though that is what happened. Very few people have looked at the internal of the pattern matcher. It has grown to become a very large and complex beast, and hard to understand.)
